WebtoHunspell/WebtoHunspell.hs

```diff
@@ -0,0 +1,1209 @@
+
+{---------}
+
+{-Custom Data Declarations (JSON).-}
+
+--Paradigm
+data Paradigm = Paradigm {
+    paradigm :: String
+  , slots :: [String]
+  , words :: A.Value
+  } deriving (Read,Show)
+
+instance FromJSON Paradigm
+    where
+        parseJSON (Object v) = Paradigm <$> (v .: "paradigm_name")
+                                        <*> (v .: "slots")
+                                        <*> (v .: "words")
+
+        parseJSON _          = mzero
+--
+
+--Inputs
+data Inputs = Inputs {
+    slotandword :: [(String,String)]
+    } deriving (Read,Show)
+--
+--
+
+{----------------------------------}
+
+{-Affix file-}
+{-Custom Data Declarations (after JSON).-}
+
+data AffixFileEncoding = 
+    AffixFileEncoding String
+    deriving (Eq,Ord,Show)
+
+data PreRuleHeader =
+    PrePfxHead String String Char
+  | PreSfxHead String String Char
+    deriving (Eq,Ord,Show) 
+
+data PreRule =
+    PrePfx String String String String
+  | PreSfx String String String String
+    deriving (Eq,Ord,Show)
+
+data RuleHeader =
+    PfxHead String String Char Int
+  | SfxHead String String Char Int
+    deriving (Eq,Ord,Show)
+
+data Rule = 
+    Pfx String String String String String 
+  | Sfx String String String String String
+    deriving (Eq,Ord,Show)
+
+{----------------------------------------}
+{------------}
+
+{-Dictionary file-}
+{-Custom Data Declarations (after JSON).-}
+
+data DictFileHeader = 
+    DictFileHeader String
+    deriving (Eq,Ord,Show)
+
+data DictFileBody = 
+    DictFileBody String
+    deriving (Eq,Ord,Show)
+
+{----------------------------------------}
+{-----------------}
+
+{-General Utility Functions.-}
+
+--linefeed -> To feed file into nested list line by line, 
+--delimited by newline characters.
+linefeed :: String -> [String]
+linefeed xs = L.lines xs
+
+--singleunnest
+singleunnest :: [a] -> a
+singleunnest [xs] = xs
+
+--tripletfst
+tripletfst :: (a,b,c) -> a
+tripletfst (a,b,c) = a
+
+--tripletsnd
+tripletsnd :: (a,b,c) -> b
+tripletsnd (a,b,c) = b
+
+--tripletthrd
+tripletthrd :: (a,b,c) -> c
+tripletthrd (a,b,c) = c
+
+--mtripletfst
+mtripletfst :: Maybe (a,b,c) -> a
+mtripletfst (Just (a,b,c)) = a
+
+--mtripletsnd
+mtripletsnd :: Maybe (a,b,c) -> b
+mtripletsnd (Just (a,b,c)) = b
+
+--mtripletthrd 
+mtripletthrd :: Maybe (a,b,c) -> c
+mtripletthrd (Just (a,b,c)) = c
+
+--tupletotriplet
+tupletotriplet :: String -> (String,String) -> (String,String,String)
+tupletotriplet x (y,z) = (x,y,z)
+
+--maptupletotriplet
+maptupletotriplet :: String -> [(String,String)] -> [(String,String,String)]
+maptupletotriplet [] []        = []
+maptupletotriplet _  []        = []
+maptupletotriplet x ((y,z):xs) = [tupletotriplet x (y,z)] ++ (maptupletotriplet x xs)
+
+{----------------------------}
+
+{-Functions to grab information from parsed JSON.-}
+
+--grabparadigm -> To grab the paradigm of the json.
+grabparadigm :: Maybe Paradigm -> String
+grabparadigm  Nothing                                  = []
```

Comment:
Could you help explain what does this line do? Is it for null data handling?

Reply:
Great question!  This function uses the Maybe monad.  This basically allows for a function to either return some value of type x, or return nothing in the case it isn't found.